code/modules/projectiles/guns/projectile/pistol.dm

```diff
@@ -237,7 +237,9 @@
 	else
 		icon_state = "pistol"
 	if(!(ammo_magazine && ammo_magazine.stored_ammo.len))
-		icon_state = "[icon_state]-e"
+		icon_state = "pistol"
```

Comment:
this will show the loaded sprite when it's empty, and won't properly show the silencer sprite

Reply:
Can't fix this one with suggestions. Silencer needs to be made into an overlay